src/dask_histogram/boost.py

```diff
@@ -34,6 +36,60 @@
 __all__ = ("Histogram", "histogram", "histogram2d", "histogramdd")
 
 
+def _hist_safe_sum(items):
+    safe_items = [item for item in items if not isinstance(item, tuple)]
+    return sum(safe_items)
+
+
+def _build_staged_tree_reduce(
```

Comment:
This "low-level" approach is fine, but DataFrameTreeReduction exists, as used for instance by AwkwardTreeReductionLayer, called in dask_awkward.core.non_trivial_reduction (i.e., it's not actually specific to dataframes)

Reply:
The only reason I didn't use DataFrameTreeReduction is because the inputs aren't from blockwise so that code doesn't work out of the box. The code I used is a directly clone of what's in dask.Bag's `reduction` function, with modifications to use the appropriate keys from all the histogram fills.

I'm happy to convert the internals to use use a more blockwise-style task labelling so we can use those tools instead, but I think that's for another PR as it would be a bigger change.

Reply:
OK then; certainly it's an improvement over previous, but we can revisit if we really need blockwise-like layers later.